text/trace/0192-messaging-semantic-conventions-spec.md

```diff
@@ -0,0 +1,659 @@
+
+##### `messaging.source.anonymous`
+
+See [`messaging.destination.anonymous`](#messagingdestinationanonymous).
+
+##### `net.app.protocol.name`
+
+The name of the underlying protocol which is used to publish and receive
+messages. This should specify application layer protocols (e. g. AMQP, MQTT, or
+HTTP) and not transport or network layer protocols (e. g. TCP, UDP, IP).
+
+See [Network Transport Attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#network-transport-attributes)
+for further details.
+
+##### `net.app.protocol.version`
+
+The version of the protocol given in [`messagingprotocolname`](#messagingprotocolname),
+if applicable.
+
+See [Network Transport Attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#network-transport-attributes)
+for further details.
+
+##### `net.peer.ip`
+
+This should be the remote address (dotted decimal for IPv4 or [RFC5952](https://datatracker.ietf.org/doc/html/rfc5952)
+of the broker or intermediary this specific message is sent to or received
+from.
+
+See [Network Transport Attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#network-transport-attributes)
+for further details.
+
+##### `net.peer.name`
+
+This should be the host name of the broker or intermediary this specific
+message is sent to or received from.
+
+See [Network Transport Attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#network-transport-attributes)
+for further details.
+
+### System-specific extensions
+
+### Examples
+
+#### Single message producer, single message push-based consumer
+
+```
+  PRODUCER                                   CONSUMER
+
+  +------------+         (link)              +------------+
+  | Publish m1 | . . . . . . . . . . . . . . | Deliver m1 |
+  +------------+                             +------------+
+```
+
+#### Single message producer, single message push-based consumer with manual settlement
+
+```
+  PRODUCER                                   CONSUMER
+
+  +------------+                             +------------------+
+  | Publish m1 | . . . . . . . . . . . . . . | Deliver m1       |
+  +------------+                             +-----+-----------++
+                                                   | Settle m1 |
+                                                   +-----------+
+```
+
+#### Single message producer, single message push-based consumer with auto-settlement
+
+```
+  PRODUCER                                   CONSUMER
+
+                                           +--------------------------+
+                                           | Ambient                  |
+  +------------+                           +-+------------+-----------+
```

Comment:
"Ambient" means the span that's active at the time of `Deliver m1`. For ex when an HTTP request arrives at a server, it may start a span before reaching the user's handler code. In this case, the user code has a "ambient" context already.

Reply:
> "Ambient" means the span that's active at the time of Deliver m1. For ex when an HTTP request arrives at a server, it may start a span before reaching the user's handler code. In this case, the user code has a "ambient" context already.

Ok thanks! It makes sense now

> About the auto-settlement I believe the ambient here has no connection, it's just to highlight a use case.

Maybe instead of having it in the title, we should have a simple description explaining the use case. This could be valid for all the others examples.